stories/index.js

```diff
@@ -1624,3 +1637,155 @@ storiesOf('InteractionLayer', module)
+        oldSeries,
+        ...params,
+      }),
+      y: staticLoader({
+        id: pair.y,
+        n: NUM_POINTS,
+        reason,
+        oldSeries,
+        ...params,
+      }),
+    };
+
+    const data = [];
+    const lastKnown = { x: undefined, y: undefined, z: undefined };
+    while (x.data.length || y.data.length) {
+      const points = {
+        x: x.data.length ? x.data[0] : { timestamp: Number.MAX_SAFE_INTEGER },
+        y: y.data.length ? y.data[0] : { timestamp: Number.MAX_SAFE_INTEGER },
+      };
+      let point;
+      if (points.x.timestamp <= points.y.timestamp) {
+        point = x.data.shift();
+        lastKnown.x = point.value;
+      }
+      if (points.y.timestamp <= points.x.timestamp) {
+        point = y.data.shift();
+        lastKnown.y = point.value;
+      }
+      lastKnown.z = point.timestamp;
+      if (
+        lastKnown.x !== undefined &&
+        lastKnown.y !== undefined &&
+        lastKnown.z !== undefined
+      ) {
+        data.push({
+          ...lastKnown,
+        });
+      }
+    }
+
+    return { data };
+  }
+  return { data: oldSeries.data };
+};
+
+storiesOf('Scatterplot', module)
+  .addDecorator(story => (
+    <div style={{ marginLeft: 'auto', marginRight: 'auto', width: '80%' }}>
+      {story()}
+    </div>
+  ))
+  .add(
+    'Scatterplot (one series)',
+    withInfo()(() => (
+      <div style={{ height: '500px', width: '500px' }}>
+        <DataProvider
+          defaultLoader={scatterplotloader}
+          baseDomain={[0, 1]}
+          series={[{ id: '1 2', color: 'steelblue' }]}
+          xAccessor={d => +d.x}
+          yAccessor={d => +d.y}
+        >
+          <Scatterplot zoomable />
+        </DataProvider>
+      </div>
+    ))
+  )
+  .add(
+    'Scatterplot (Geometry series)',
+    withInfo()(() => (
+      <div style={{ height: '100vh', width: '100%' }}>
+        <DataProvider
+          defaultLoader={scatterplotloader}
+          baseDomain={[0, 1]}
+          series={[
+            { id: 'sincos', color: '#ACF39D' },
+            { id: 'sintan', color: '#E85F5C' },
+            { id: 'pow', color: '#9CFFFA' },
+          ]}
+          xAccessor={d => +d.x}
+          yAccessor={d => +d.y}
+        >
+          <Scatterplot zoomable />
+        </DataProvider>
+      </div>
+    ))
+  )
+  .add(
+    'Scatterplot (many pairs)',
+    withInfo()(() => (
+      <DataProvider
+        defaultLoader={scatterplotloader}
+        baseDomain={[0, 1]}
+        series={[
+          { id: '1 2', color: '#ACF39D' },
+          { id: '2 3', color: '#E85F5C' },
+          { id: '3 4', color: '#9CFFFA' },
+          { id: '4 5', color: '#773344' },
+          { id: '5 6', color: '#E3B5A4' },
+          { id: '6 7', color: '#2E0219' },
+          { id: '7 8', color: '#2E0219' },
+          { id: '8 9', color: '#2E0219' },
+        ]}
+        xAccessor={d => +d.x}
+        yAccessor={d => +d.y}
+      >
+        <Scatterplot height={500} width={500} zoomable />
```

Comment:
It's surprising to me that even with fixed sizes the yAxis for this and the next demo keeps growing. Would you mind explain a bit why?

Reply:
I think there was like a few-px difference (because of errant whitespace or something?) I set the height of the scatterplot to `height - 50` but never set the x-axis to `50`. Anyway, I removed the ability to set width/height here and instead just have it always be the size of its container.